lookout/style/typos/analyzer.py

```diff
@@ -128,34 +136,85 @@ def generate_typos_fixes(self, changes: Sequence[Change]) -> Iterator[TypoFix]:
                 if not new_identifiers:
                     continue
                 suggestions = self.check_identifiers([n.token for n in new_identifiers])
+                if not suggestions:
+                    continue
                 for index in suggestions.keys():
-                    corrections = suggestions[index]
-                    for token in corrections.keys():
-                        yield TypoFix(
-                            head_file=file,
-                            token=new_identifiers[index].token,
-                            candidates=corrections[token],
-                            line_number=new_identifiers[index].start_position.line,
-                        )
-
-    def render_comment_text(self, typo_fix: TypoFix) -> str:
+                    yield TypoFix(
+                        head_file=file,
+                        identifier=new_identifiers[index].token,
+                        candidates={token: [Candidate(*sugg) for sugg in suggestions[index][token]]
+                                    for token in suggestions[index]},
+                        line_number=new_identifiers[index].start_position.line,
+                    )
+
+    def render_comment_text(self, typo_fix: TypoFix) -> (str, Iterable[float]):
         """
         Generate the text of the comment for the specified typo fix.
 
         :param typo_fix: Information about typo fix required to render a comment text.
         :return: string with the generated comment.
         """
-        # TODO: move this logic to template. See FormatAnalyzer.render_comment_text()
-        corrections_line = ", ".join("%s (%d%%)" % (
-            c.token, int(c.confidence * 100)) for c in typo_fix.candidates)
-        return "Possible typo in \"%s\". Suggestions: %s" % (typo_fix.token, corrections_line)
+        file_lines = typo_fix.head_file.content.decode("utf-8", "replace").splitlines()
+        suggestions, confidences = [], []
+        for final_sugg, conf in self.generate_identifier_suggestions(
+                typo_fix.candidates, typo_fix.identifier,
+        ):
```

Comment:
I don't think so. 
You mentioned `_get_comment_confidence` in comments above - and it was called in `render_comment_text` - so it's a logic specific for comment generation. As everything inside `render_comment_text` is a logic specific for the generation of specific types of comments (1 loop only now).
`typos_fixes` already contains all the required information to generate everything. But obviously, it contains only generic information.
Before it was subtoken to subtoken suggestions. Now - identifier to identifier.
And it shouldn't be bloated with non-generic information IMHO because it will increase the complexity of very `lookout-specific` function `generate_typos_fixes` and we will lose in flexibility and extendability.

Reply:
Sorry for being unclear, @EgorBu, the point is that I need to call this logic from the reporter to get identifiers suggestion if I do not know what do we exactly recommend. Let's agree that reporter should not do it even if it is two lines logic.

So, the `analyze` function split to `generate_typos_fixes` and `render_comment_text` was by design created to have everything that reporter need in `TypoFix`. That why I asked to put in inside `generate_typos_fixes()`.